pkg/config/loader.go

```diff
@@ -0,0 +1,818 @@
+			continue
+		}
+		log.Debug("atmos merged config file", "path", configPath)
+
+		// Process Imports and Deep Merge
+		if err := cl.processConfigImports(); err != nil {
+			log.Debug("error processing imports after config file", "path", configPath, "error", err)
+			continue
+		}
+		err = cl.deepMergeConfig()
+		if err != nil {
+			log.Debug("error merge config file", "path", configPath, "error", err)
+		}
+	}
+
+	return nil
+}
+
+func parseArraySeparator(s string) []string {
+	var result []string
+	for _, part := range strings.Split(s, ";") {
+		part = strings.TrimSpace(part)
+		if part != "" {
+			result = append(result, part)
+		}
+	}
+	return result
+}
+
+func (cl *ConfigLoader) loadAtmosConfigFromEnv(atmosCliConfigEnv string) error {
+	atmosCliConfigEnvPaths := parseArraySeparator(atmosCliConfigEnv)
+	if len(atmosCliConfigEnvPaths) == 0 {
+		return fmt.Errorf("ATMOS_CLI_CONFIG_PATH is not a valid paths")
+	}
+	if atmosCliConfigEnv == AtmosYamlFuncGitRoot {
+		gitRoot, err := u.GetGitRoot()
+		if err != nil {
+			return fmt.Errorf("failed to resolve base path from `!repo-root` Git root error: %w", err)
+		}
+		log.Debug("atmos config from ATMOS_CLI_CONFIG_PATH !repo-root Git root dir", "path", gitRoot)
+		atmosCliConfigEnv = gitRoot
+	}
+	var atmosFilePaths []string
+	for _, configPath := range atmosCliConfigEnvPaths {
+		atmosFilePath, err := cl.getPathAtmosCLIConfigPath(configPath)
+		if err != nil {
+			log.Debug("failed to find config file", "path", atmosFilePath, "error", err)
+			continue
+		}
+		atmosFilePaths = append(atmosFilePaths, atmosFilePath...)
+	}
+	if len(atmosFilePaths) == 0 {
+		return fmt.Errorf("Failed to find config files in ATMOS_CLI_CONFIG_PATH '%s'", atmosCliConfigEnv)
+	}
+	for _, atmosFilePath := range atmosFilePaths {
+		log.Debug("atmos config from ATMOS_CLI_CONFIG_PATH ", "path", atmosFilePath)
+		err := cl.loadConfigFileViber(cl.atmosConfig, atmosFilePath, cl.viper)
+		if err != nil {
+			log.Debug("error processing config file", "path", atmosFilePath, "error", err)
+			continue
+		}
+		err = cl.deepMergeConfig()
+		if err != nil {
+			log.Debug("error merge config file", "path", atmosFilePath, "error", err)
+			continue
+		}
+		log.Debug("atmos merged config file", "path", atmosFilePath)
+		// Process Imports and Deep Merge
+		if err := cl.processConfigImports(); err != nil {
+			log.Debug("atmos config file found on ATMOS_CLI_CONFIG_PATH", "path", atmosFilePath, "error", err)
+			continue
+		}
+		err = cl.deepMergeConfig()
+		if err != nil {
+			log.Debug("error merge config file", "path", atmosFilePath, "error", err)
+		}
+	}
+	return nil
+}
+
+func (cl *ConfigLoader) getPathAtmosCLIConfigPath(atmosCliConfigPathEnv string) ([]string, error) {
+	isDir, err := u.IsDirectory(atmosCliConfigPathEnv)
+	if err != nil {
+		if err == os.ErrNotExist {
+			return nil, fmt.Errorf("error ATMOS_CLI_CONFIG_PATH: %w", err)
+		}
+		return nil, fmt.Errorf("error ATMOS_CLI_CONFIG_PATH: %w", err)
+	}
+	if !isDir {
```

Comment:
But we should be able to specify both a directory or a file, no? E.g. the path to `atmos.d` directory